### PR TITLE
Adding ability to generate raw insert sql on persistable objects

### PR DESCRIPTION
**Pull request by srksag-zz**

### CODE REVIEW

File: components/persistor/lib/knex/db.ts [hunk 1]
```diff
@@ -982,6 +990,37 @@ module.exports = function (PersistObjectTemplate) {
...
+                if (!schema || !schema.parents || !schema.parents[prop] || !schema.parents[prop].id)
+                    throw new Error(template.__name__ + '.' + prop + ' is missing a parents schema entry');
+                var foreignKey = (schema.parents && schema.parents[prop]) ? schema.parents[prop].id : prop;
+                data[foreignKey] = obj[prop];
+            }
+            else 
+                data[prop] = obj[prop];
+        }
+
+        return knex.insert(data).toString();
```
Comment (sshankar19) [inline components/persistor/lib/knex/db.ts:1021]: This just generates as a string correct? should we make it possible to pass the query builder object back?

File: components/persistor/lib/knex/db.ts [hunk 2]
```diff
@@ -892,6 +892,14 @@ module.exports = function (PersistObjectTemplate) {
         return map;
     }
 
+    function generateNotifyQueries(template, queriesToNotify, query) {
```
Comment (sshankar19) [inline components/persistor/lib/knex/db.ts:895]: Can we have some types or params as to what is going on in this function?

File: components/persistor/lib/knex/db.ts (hunk 1 above)
Comment (srksag-zz) [reply components/persistor/lib/knex/db.ts:1021]: Thought about that approach, but the current usecase is making the consumer write loops and all if we return query builder, we want to track them in a file backup and run all them once.

File: components/persistor/lib/knex/db.ts (hunk 2 above)
Comment (srksag-zz) [reply components/persistor/lib/knex/db.ts:895]: this is an internal function, I can type second and third parameter, but first one is a supertype, may be we can consider this as a bigger refactor.

**Review by rognorak — APPROVED** [review]

**Review by nsaenz7 — APPROVED** [review]

**Pull request merged**